### PR TITLE
 Phase 2 in AddBlock Function

### PR DESCRIPTION
**Pull request by arosso3**
#66

### CODE REVIEW

File: producer/src/blockchain/blockchain.go [hunk 1]
```diff
@@ -33,30 +34,51 @@ func AddBlock(b block.Block, filename string, databaseName string) error { // Ad
...
+	database, err := sql.Open("sqlite3", databaseName)
+	// Checks if the opening was successful
+	if err != nil {
+		return err
+	}
 
-	// statement, _ := database.Prepare("CREATE TABLE IF NOT EXISTS metadata (height INTEGER PRIMARY KEY, position INTEGER, size INTEGER, hash TEXT)")
-	// statement.Exec()
-	// statement, _ = database.Prepare("INSERT INTO")
+	statement, err := database.Prepare("CREATE TABLE IF NOT EXISTS metadata (height INTEGER PRIMARY KEY, position INTEGER, size INTEGER, hash TEXT)")
```
Comment (kstola2) [inline producer/src/blockchain/blockchain.go]: I'm not sure we need this called each time, we can assume at this point that the table is created.

File: producer/src/blockchain/blockchain.go [hunk 2]
```diff
@@ -33,30 +34,51 @@ func AddBlock(b block.Block, filename string, databaseName string) error { // Ad
...
+	}
+	_, err = statement.Exec()
+	if err != nil {
+		return err
+	}
+	statement, err = database.Prepare("INSERT INTO metadata (height, position, size, hash) VALUES (?, ?, ?, ?)")
+	if err != nil {
+		return err
+	}
+	_, err = statement.Exec(b.Height, bPosition, bLen, b.PreviousHash)
```
Comment (kstola2) [inline producer/src/blockchain/blockchain.go]: The hash should be the Hash of b, not `b.PreviousHash`

File: producer/src/blockchain/blockchain.go (hunk 1 above)
Comment (arosso3) [reply producer/src/blockchain/blockchain.go]: Do you want to get rid of it or just have it run when its empty?

Comment (kstola2) [reply producer/src/blockchain/blockchain.go]: I think we can get rid of this particular statement yeah

Comment (arosso3) [reply producer/src/blockchain/blockchain.go]: Ok, but then it fails the test because there is no table to add to

Comment (kstola2) [reply producer/src/blockchain/blockchain.go]: The table created in the test doesn't persist into the function call?

Comment (arosso3) [reply producer/src/blockchain/blockchain.go]: The first test doesn't create a table

Comment (kstola2) [reply producer/src/blockchain/blockchain.go]: Ah okay. Let's fix that

Comment (kstola2) [reply producer/src/blockchain/blockchain.go]: Just pushed to your branch, let me know if that fixes it

Comment (kstola2) [reply producer/src/blockchain/blockchain.go]: Had to push once more to remove the DB at the end of the first test apologies

**Comment by arosso3** [comment]
Updated function

**Pull request merged**